geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java

```diff
@@ -1142,7 +1142,7 @@ public void createMappingWithExistingQueueFails() {
             + " must not already exist.");
   }
 
-  private static class Employee implements PdxSerializable {
+  public static class Employee implements PdxSerializable {
```

Comment:
In the test, `CreatemappingPreconditionCheckFunction` is called and executed on several JVMs(Distributed members).
In this Function, PDX class such as Employee is instantiated, so I need to be able to access this class even if it runs on another JVM.
If it is in private scope, the following exceptions will be thrown

```
[error 2020/11/07 12:08:19.411 JST <Function Execution Processor2> tid=69] Could not generate a PdxType for the class org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest$IdAndName because it did not have a public zero arg constructor. Details: java.lang.IllegalAccessException: Class org.apache.geode.connectors.jdbc.internal.JdbcConnectorServiceImpl can not access a member of class org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest$IdAndName with modifiers "public"
org.apache.geode.connectors.jdbc.JdbcConnectorException: Could not generate a PdxType for the class org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest$IdAndName because it did not have a public zero arg constructor. Details: java.lang.IllegalAccessException: Class org.apache.geode.connectors.jdbc.internal.JdbcConnectorServiceImpl can not access a member of class org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest$IdAndName with modifiers "public"
    at org.apache.geode.connectors.jdbc.internal.JdbcConnectorServiceImpl.createInstance(JdbcConnectorServiceImpl.java:361)
    at org.apache.geode.connectors.jdbc.internal.JdbcConnectorServiceImpl.generatePdxTypeForClass(JdbcConnectorServiceImpl.java:334)
    at org.apache.geode.connectors.jdbc.internal.JdbcConnectorServiceImpl.getPdxTypeForClass(JdbcConnectorServiceImpl.java:320)
    at org.apache.geode.connectors.jdbc.internal.cli.CreateMappingPreconditionCheckFunction.executeFunction(CreateMappingPreconditionCheckFunction.java:74)
    at org.apache.geode.management.cli.CliFunction.execute(CliFunction.java:37)
    ...
```

Other similar tests like CreateMapingCommandForProxyRegionDUnitTest are also public scope.

Reply:
On the surface, it seems a trivial change from `private` to `public`, in order to fix the exception.  However, under the hood, it is doing unnecessary serialization and deserialization of the `Employee` class. Because `getPdxTypeForClass()`, `generatePdxTypeForClass() ` and `createInstance()` are moved from `CreateMappingPreconditionCheckFunction` to `JdbcConnectorServiceImpl`, which is probably because you want to use these methods in `RegionMappingConfiguration`.

I would recommend keeping `getPdxTypeForClass()`, `generatePdxTypeForClass() ` and `createInstance()` in `CreateMappingPreconditionCheckFunction`. And in `RegionMappingConfiguration`, implement similar methods. Though it looks like duplicating the code, it does avoid unnecessary serialization and deserialization of user defined classes like `Employee`, which is more important.

And I don't think adding more methods to `JdbcConnectorService` is necessary. e.g. `createDefaultFieldMapping()` is something you want to automatically achieve for the users, so it is not necessary be exposed as an API. Just put it as a private method in `RegionMappingConfiguration` should be good. The same reason applies for `getTableMetaDataView()` and `getPdxTypeForClass`.  Basically, keep `JdbcConnectorService` unchanged.

Reply:
Thank you for your detailed explanation. I have reverted the `CreateMappingPreconditionCheckFunction` and `JdbcConnectorService` to the original source code. And I have implemented similar methods in `RegionMappingConfiguration`.